app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt

```diff
@@ -153,13 +147,9 @@ class MigrationManager @Inject constructor(
         return resultAcc.toMap()
     }
 
-    private fun migrationFailure(failure: CoreFailure): MigrationData.Result.Failure.Type = when (failure) {
-        is NetworkFailure.NoNetworkConnection -> MigrationData.Result.Failure.Type.NO_NETWORK
-        is StorageFailure.DataNotFound -> MigrationData.Result.Failure.Type.DATA_NOT_FOUND
-        is ServerConfigFailure.UnknownServerVersion -> MigrationData.Result.Failure.Type.UNKNOWN_SERVER_VERSION
-        is ServerConfigFailure.NewServerVersion -> MigrationData.Result.Failure.Type.TOO_NEW_VERSION
-        is MigrationFailure.InvalidRefreshToken -> MigrationData.Result.Failure.Type.UNKNOWN
-        else -> MigrationData.Result.Failure.Type.UNKNOWN
+    private fun migrationFailure(failure: CoreFailure): MigrationData.Result = when (failure) {
+        is NetworkFailure.NoNetworkConnection -> MigrationData.Result.Failure
+        else -> MigrationData.Result.Success
```

Comment:
If the failure is different then `Network`, should return success as well ?

Reply:
We discussed that already and for the migration, if it failed for some reason other than network issue (missing or corrupted data from scala app) then we can't do anything about it as running the migration again won't fix it, so we just try to migrate as many data as we can and finish. 
Mojtaba already suggested that if something goes wrong, we should show the user a notification so then he knows that there was an error that the app  was unable to resolve on its own and he should login again on that account. I can do it in another PR.